tests/unit/CMakeLists.txt

```
@@ -32,7 +32,7 @@ function(add_unit_test MAIN_SOURCE)
   target_compile_definitions(${BASENAME} PRIVATE ${UNIT_COMPILE_DEFINITIONS})
   add_symlink_to_built_opencl_dynlib(${BASENAME})
 
-  target_link_libraries(${BASENAME} PRIVATE ${POCL_LIBRARY_NAME})
+  target_link_libraries(${BASENAME} PRIVATE ${POCL_LIBRARY_NAME} ${POCLU_LINK_OPTIONS})
```

It would be better if the link options were not passed in via `target_link_libraries` due to reasons in [CMake docs](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) (generally considered unsafe, placement of the options may not be correct).
```suggestion
  target_link_libraries(${BASENAME} PRIVATE ${POCL_LIBRARY_NAME})
  target_link_options(${BASENAME} PRIVATE ${POCLU_LINK_OPTIONS})
```